proposals/vector_buffer.json

```diff
@@ -0,0 +1,59 @@
+{
+    "id": "vector_buffer",
+    "summary": "Buffer geometries by distance",
+    "description": "Buffers each input geometry by a given distance, which can either expand (dilate) or a shrink (erose) the geometry. Buffers can be applied to points, lines and polygons, but the results are always polygons. Multi-part types (e.g. `MultiPoint`) are also allowed.",
+    "categories": [
+        "vector"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "geometries",
+            "description": "Geometries to apply the buffer on. Vector properties are preserved for vector data cubes and all GeoJSON Features.\n\nTo maximize interoperability, a nested `GeometryCollection` should be avoided. Furthermore, a `GeometryCollection` composed of a single type of geometries should be avoided in favour of the corresponding multi-part type (e.g. `MultiPolygon`).",
+            "schema": [
+                {
+                    "type": "object",
+                    "subtype": "geojson"
+                },
+                {
+                    "type": "object",
+                    "subtype": "vector-cube"
+                }
+            ]
+        },
+        {
+            "name": "distance",
+            "description": "The distance of the buffer in the unit given by the `unit` parameter. A positive distance expands the geometries and results in outward buffering (dilation) while a negative distance shrinks the geometries and results in inward buffering (erosion).",
+            "schema": {
+                "type": "number",
+                "not": {
+                    "const": 0
+                }
+            }
+        },
+        {
+            "name": "unit",
+            "description": "The unit the distance is given in, either in the unit of the spatial reference system (`null`, the default value) or in meters (`m`).",
+            "optional": true,
+            "default": null,
+            "schema": [
+                {
+                    "title": "Unit of the SRS",
+                    "type": "null"
+                },
+                {
+                    "title": "Meters",
+                    "type": "string",
+                    "const": "m"
```

Comment:
Correct me if am wrong but isn't it better to always use the units of the current CRS there? Otherwise I assume that you'd need to do a reprojection or some other kind of conversion on the fly, right? Is that planned?

Reply:
I guess the reasoning is that GeoJSON only strictly supports epsg:4326, while you usually want to express the buffer in meters?

Reply:
Exactly, geoJSON officially supports only epsg:4326. Having also meters would imply to convert it to degrees, but we should inform the user that it's not a perfect conversion.

Reply:
Well, this also supports vector-cube, which are not restricted to GeoJSON and EPSG4326. So @mattia6690 is raising a valid point here. Until now, we have two different approaches in processes:
1. in resample_spatial it is based on the unit of the CRS: "Specified in the units of the target projection."
2. in apply_neighborhood you can either specify "m" or "px" as the unit for the size.

What is preferable here? The easiest solution would probably be 1.

Reply:
We've been doing geometry buffering in a couple of VITO use cases already, and I think the most natural way for the user to specify this is in meter  (e.g. inward buffering to eliminate edge issues). Right, @jdries ?
So I think option 2 makes most sense.

Option 1 assumes that your have a target data cube and that its CRS is sensible to express the buffer distance in, which is not necessarily try

Reply:
I'm afraid it's even more difficult.
A vector cube normally has a projection system associated with it, which can be geodetic (lat lon) or specified in meters.
The meters of a projected coordinated system however, do not necessarily correspond to meters on earth.
IMO the most logical buffering operation indeed uses meters, but on the earth. This is however not the easiest variant.

You could also consider supporting the other variants, but then we indeed need the necessary properties to differentiate.

Reply:
Hmm, so I guess the "easiest" way is to just redirect these issues to the underlying libraries? It seems most rely on the "unit of the spatial reference system, so we can probably just start with that? Updated the process accordingly.

Reply:
> just redirect these issues to the underlying libraries? It seems most rely on the "unit of the spatial reference system, so we can probably just start with that? 

I don't think that is very practical. E.g. if you start from GeoJSON, which is strictly lat-lon, you have to specify the buffer in degrees, which no user probably wants to do in a real use case. We also don't have a vector reprojection process, so you're stuck.

Reply:
Right now every process does something differently, some use meters, some degrees, some "unit of the srs", etc. We need to commit to a consistent behavior, I think.